mscore/startupWizard.cpp

```diff
@@ -0,0 +1,90 @@
+                 << QObject::tr("T1 - Standard German")
+                 << QObject::tr("T2")
+                 << QObject::tr("Swiss German")
+                 << QObject::tr("Belgian French")
+                 << QObject::tr("Canadian French")
+                 << QObject::tr("French French")
+                 << QObject::tr("Swiss French")
+                 << QObject::tr("Spanish")
+                 << QObject::tr("Italian")
+                 << QObject::tr("Portuguese Brazilian ABNT")
+                 << QObject::tr("Portuguese")
+                 << QObject::tr("US-International")
+                 << QObject::tr("Dutch Layout")
+                 << QObject::tr("Dubeolsik (두벌식)")
+                 << QObject::tr("Sebeolsik (세벌식)")
+                 << QObject::tr("OTHER");
+
+        QString labelMessage = QObject::tr("Enter your keyboard layout");
+        QString result = QInputDialog::getItem(new QWidget(), "Startup Wizard", labelMessage, layoutList);
+        return result;
+      }
+
+void StartupWizard::autoSelectShortcuts(QString keyboardLayout)
+      {
+      QMap<QString, QString> *layoutToShortcut = new QMap<QString, QString>();
+      layoutToShortcut->insert("US-QWERTY", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("UK-QWERTY", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("QWERTZ", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("AZERTY", ":/data/shortcuts_AZERTY.xml");
+      layoutToShortcut->insert("BEPO", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("DVORAK", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("JIS - Standard Japanese", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("OYAYUBI SHIFUTO", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("T1 - Standard German", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("T2", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Swiss German", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Belgian French", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Canadian French", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("French French", ":/data/shortcuts_AZERTY.xml");
+      layoutToShortcut->insert("Swiss French", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Spanish", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Italian", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Portuguese Brazilian ABNT", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Portuguese", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("US-International", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Dutch Layout", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Dubeolsik (두벌식)", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("Sebeolsik (세벌식)", ":/data/shortcuts.xml");
+      layoutToShortcut->insert("OTHER", ":/data/shortcuts.xml");
+
+      QString fileLocation = layoutToShortcut->value(keyboardLayout);
```

Comment:
Also we do have 2 lists, which need to be kept in sync. Probably better to have just one.